### PR TITLE
Fix bad assumption that <use> can override target fill

### PR DESCRIPTION
**Pull request by rsheeter**
Test to expose and prove fix works

Fixes https://github.com/googlefonts/nanoemoji/issues/337

### CODE REVIEW

File: src/nanoemoji/svg.py [hunk 1]
```diff
@@ -340,7 +355,57 @@ def _apply_paint(
...
+    svg_defs = svg.xpath_one("//svg:defs")
+
+    if reused_el in svg_defs:
+        return  # nop
+
+    svg_use = etree.Element("use", nsmap=svg.svg_root.nsmap)
+    svg_use.attrib[_XLINK_HREF_ATTR_NAME] = f"#{reuse_result.glyph_name}"
+    # if reused_el hasn't been given a parent yet just let the <use> replace it
+    # otherwise move it from current to new parent
+    if reused_el.getparent() is None:
```
Comment (anthrotype) [inline src/nanoemoji/svg.py:380]: when does reused_el have no parent? I don't think _migrate_to_defs ever meets a parent-less reused_el.
IIUC, at a first occurrence of a paint glyph, when by definition there is no (not yet) reuse, we always append the corresponding element to the parent element (svg root or g), thus by the time a reuse is found, that reused element already got a parent.
If that's the case, I think here we should assert/assume that reused_el already has a parent.

**Review by anthrotype — APPROVED** [review]
Looks good, thank you!

File: src/nanoemoji/svg.py (hunk 1 above)
Comment (rsheeter) [reply src/nanoemoji/svg.py:380]: Hm. It was definitely possible at some point but now I'm failing to come up with when that would be. I think I'm going to file a new ticket for that and leave it for this one.

Comment (rsheeter) [reply src/nanoemoji/svg.py:380]: https://github.com/googlefonts/nanoemoji/issues/346

**Pull request merged**